rbc/tests/test_omnisci_udtf.py

```diff
@@ -1,49 +1,140 @@
 import pytest
-
-
-rbc_omnisci = pytest.importorskip('rbc.omniscidb')
-available_version, reason = rbc_omnisci.is_available()
-if available_version and available_version >= (5, 4):
-    reason = ('Old-style UDTFs are available for omniscidb 5.3 or older, '
-              'currently connected to omniscidb '
-              + '.'.join(map(str, available_version)))
-    available_version = ()
-pytestmark = pytest.mark.skipif(not available_version, reason=reason)
+from rbc.tests import omnisci_fixture
+import numpy as np
 
 
 @pytest.fixture(scope='module')
 def omnisci():
-    config = rbc_omnisci.get_client_config(debug=not True)
-    m = rbc_omnisci.RemoteOmnisci(**config)
-    table_name = 'rbc_test_omnisci_udtf'
-    m.sql_execute('DROP TABLE IF EXISTS {table_name}'.format(**locals()))
-
-    sqltypes = ['FLOAT', 'DOUBLE', 'TINYINT', 'SMALLINT', 'INT', 'BIGINT',
-                'BOOLEAN']
-    colnames = ['f4', 'f8', 'i1', 'i2', 'i4', 'i8', 'b']
-    table_defn = ',\n'.join('%s %s' % (n, t)
-                            for t, n in zip(sqltypes, colnames))
-    m.sql_execute(
-        'CREATE TABLE IF NOT EXISTS {table_name} ({table_defn});'
-        .format(**locals()))
+    for o in omnisci_fixture(globals()):
+        define(o)
+        yield o
+
+
+def define(omnisci):
+
+    T = ['int64', 'int32', 'int16', 'int8', 'float64', 'float32']
```

Comment:
I suppose you don't really need to test all combinations of types given you're testing if function combination works in omniscidb.

Reply:
true, but one never knows what will break :) Skipping some types will give only a minor decrease in testing time.